pkg/controllers/multicluster/endpointslice_plan.go

```diff
@@ -39,6 +39,11 @@ type EndpointSlicePlan struct {
 	ClusterId string
 }
 
+func (p *EndpointSlicePlan) CheckIPType() model.IPType {
+	// Peek at the first endpoint for its IPType. All endpoints in a slice will be of the same IPType.
+	return p.Desired[0].IPType
```

Comment:
Not currently, but I'll add a fallback to IPV4 in case a future change breaks this. 

https://github.com/aws/aws-cloud-map-mcs-controller-for-k8s/blob/main/pkg/controllers/multicluster/cloudmap_controller.go#L102

Reply:
When we implement dual stack support, we can improve the way the address type is calculated/tracked for a slice, etc.